lib/itamae/resource/file.rb

```diff
@@ -169,6 +176,34 @@ def send_tempfile
           f.unlink if f
         end
       end
+
+      def backup
+        return if !attributes.backup || attributes.backup <= 0
+
+        savetime = Time.now.strftime('%Y%m%d%H%M%S')
+        backup_filename = "#{attributes.path}.itamae-#{savetime}"
+        backup_path = ::File.join(BACKUP_PATH, backup_filename)
+        backup_directory = ::File.dirname(backup_path)
+
+        run_specinfra(:create_file_as_directory, backup_directory)
```

Comment:
I have a concern about permission. For instance, the following situation can confuse users:

1. User A runs recipe A
    1. Recipe A creates a backup file of `/etc/foo.conf`
    1. `/var/itamae/backup/etc` is mode 755 and owned by user A
1. User B runs recipe B
    1. Recipe A creates a backup file of `/etc/bar.conf`
    1. User B can't create a directory or a file in `/var/itamae/backup/etc` because it is owned by user A

WDYT?

I think it is better all directories under `/var/lib/itamae` are `755`